testing/events_test.go

```diff
@@ -0,0 +1,223 @@
+package ibctesting_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	abci "github.com/cometbft/cometbft/abci/types"
+
+	"github.com/cosmos/ibc-go/v8/modules/core/02-client/types"
+	channeltypes "github.com/cosmos/ibc-go/v8/modules/core/04-channel/types"
+	ibctesting "github.com/cosmos/ibc-go/v8/testing"
+)
+
+func TestParsePacketsFromEvents(t *testing.T) {
+	tests := []struct {
+		name            string
+		events          []abci.Event
+		expectedPackets []channeltypes.Packet
+		expectedError   string
+	}{
+		{
+			name:          "fail: no events",
+			events:        []abci.Event{},
+			expectedError: "ibctesting.ParsePacketsFromEvents: acknowledgement event attribute not found",
```

Comment:
Since we are using `ErrorContains` it could be enough to just have this (same for the other failure test cases):

```suggestion
			expectedError: "acknowledgement event attribute not found",
```